src/api/mod.rs

```diff
@@ -115,3 +137,154 @@ fn build_fetch_query<T: Serialize>(path: &str, form: T) -> String {
+    .dyn_into::<web_sys::HtmlDocument>()
+    .map_err(|_| LemmyAppError::APIError {
+      error: String::from("DOM document could not be cast"),
+    })?
+    .set_cookie(&wasm_cookies::cookies::set(path, value, &CookieOptions::default())[..])
+    .map_err(|_| LemmyAppError::APIError {
+      error: String::from("Cookie could not be set"),
+    });
+
+  r
+}
+
+#[cfg(not(feature = "ssr"))]
+pub async fn remove_cookie_wrapper(path: &str) -> Result<(), LemmyAppError> {
+  use leptos::window;
+  use wasm_bindgen::JsCast;
+
+  let r = window()
+    .document()
+    .ok_or(LemmyAppError::APIError {
+      error: String::from("DOM document is None"),
+    })?
+    .dyn_into::<web_sys::HtmlDocument>()
+    .map_err(|_| LemmyAppError::APIError {
+      error: String::from("DOM document could not be cast"),
+    })?
+    .set_cookie(&wasm_cookies::cookies::delete(path))
+    .map_err(|_| LemmyAppError::APIError {
+      error: String::from("Cookie could not be set"),
+    });
+
+  r
+}
+
+#[cfg(feature = "ssr")]
+pub async fn set_cookie_wrapper(path: &str, value: &str) -> Result<(), LemmyAppError> {
+  use actix_web::{
+    cookie::{
+      time::{Duration, OffsetDateTime},
+      Cookie,
+    },
+    http::{header, header::HeaderValue},
+  };
+  use leptos::expect_context;
+  use leptos_actix::ResponseOptions;
+
+  let response = expect_context::<ResponseOptions>();
+
+  let mut cookie = Cookie::build(path, value).finish();
+  let mut now = OffsetDateTime::now_utc();
+  now += Duration::weeks(1);
+  cookie.set_expires(now);
+  cookie.set_path("/");
+
+  if let Ok(cookie) = HeaderValue::from_str(&cookie.to_string()) {
+    response.insert_header(header::SET_COOKIE, cookie);
+  }
+
+  Ok(())
+}
+
+#[cfg(feature = "ssr")]
+#[allow(dead_code)]
+pub async fn remove_cookie_wrapper(path: &str) -> Result<(), LemmyAppError> {
+  use actix_web::{
+    cookie::{
+      time::{Duration, OffsetDateTime},
+      Cookie,
+    },
+    http::{header, header::HeaderValue},
+  };
+  use leptos::expect_context;
+  use leptos_actix::ResponseOptions;
+
+  let response = expect_context::<ResponseOptions>();
+
+  let mut cookie = Cookie::build(path, "").finish();
+  let mut now = OffsetDateTime::now_utc();
+  now += Duration::weeks(-1);
+  cookie.set_expires(now);
+  cookie.set_path("/");
+
+  if let Ok(cookie) = HeaderValue::from_str(&cookie.to_string()) {
+    response.insert_header(header::SET_COOKIE, cookie);
+  }
+
+  Ok(())
+}
+
+#[cfg(feature = "ssr")]
+pub async fn get_cookie_wrapper(path: &str) -> Result<String, LemmyAppError> {
+  use actix_web::HttpRequest;
+  use leptos_actix::extract;
+
+  let path_string = path.to_string().clone();
+
+  let cookie_value = extract(|req: HttpRequest| async move {
+    if let Some(c) = req.cookie(&path_string) {
+      let s = c.clone();
+      s.value().to_string()
+    } else {
+      "".to_string()
+    }
+  })
+  .await
+  .map_err(|e| LemmyAppError::APIError {
+    error: e.to_string(),
+  })?;
+
+  Ok(cookie_value)
+}
```

Comment:
There is an [actix-session crate](https://docs.rs/actix-session/latest/actix_session/) that should simplify handling cookies. It will allow you to set up a middleware that lets you extract cookies in a form that's easier to user that what is being done currently.

Reply:
maybe we can refactor this away later as part of a cookie oriented refactoring?

Reply:
IMO its perfectly fine to leave this cookie stuff in here, we can always remove it at a later date.